src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -1163,6 +1163,28 @@ export default class DefaultDeviceController
     return device && device.id ? device : null;
   }
 
+  private hasSameMediaStreamId(
+    kind: string,
+    selection: DeviceSelection,
+    proposedConstraints: MediaStreamConstraints
+  ): boolean {
+    // Checking for stream using the fake constraint created in calculateMediaStreamConstraints
+    let streamId;
+    if (kind === 'audio') {
+      // @ts-ignore
+      streamId = proposedConstraints?.audio.streamId;
+      /* istanbul ignore next */
+      // @ts-ignore
+      return streamId && streamId === selection.constraints?.audio?.streamId;
```

Comment:
The return type of this function is `boolean` but it may return `undefined`. You can change the first check to:
```
!!streamId &&
```